packages/graphql/src/translate/where/create-node-where-and-params.ts

```diff
@@ -60,11 +62,12 @@ function createNodeWhereAndParams({
             DISTANCE: "=",
         };
 
-        const re = /(?<field>[_A-Za-z][_0-9A-Za-z]*?)(?:_(?<not>NOT))?(?:_(?<operator>INCLUDES|IN|MATCHES|CONTAINS|STARTS_WITH|ENDS_WITH|LT|GT|GTE|LTE|DISTANCE))?$/gm;
+        const re =
+            /(?<field>[_A-Za-z][_0-9A-Za-z]*?)(?:_(?<not>NOT))?(?:_(?<operator>INCLUDES|IN|MATCHES|CONTAINS|STARTS_WITH|ENDS_WITH|LT|GT|GTE|LTE|DISTANCE))?$/gm;
 
         const match = re.exec(key);
 
-        const fieldName = match?.groups?.field;
+        const fieldName = match?.groups?.field as string;
```

Comment:
**suggestion (blocking-ish)**: mmm... I think is great we are adding typings to existing code, but I'm worried, the typings here are lying, as `fieldName` is actually `string | undefined`. If we are **sure** match and groups exist, I would rather use `match!.groups!.field` instead of casting to string (which may silently fail).

If, however, we are trying to handle the case of `match?.groups?` being `undefined`, I think it is better to explicitly define what should happen if `fieldName` is undefined:
```suggestion
        const fieldName = match?.groups?.field;
        if(fieldName===undefined){
            throw new Error(...)
        }
```

Reply:
This is an interesting one! We have 100% confidence in the match, because we own and control the input.

However, `match!.groups!.field` is currently an ESLint violation (null assertion) - is this a rule we want to disable perhaps?

Reply:
I'm not sure tbh, if we want to keep the rule (which somehow makes sense) I think we could go with some `assertString` helper, that returns the proper type and fails if it is not a string.

My point is that this should fail if it is not a string imo

Reply:
I guess the thing for me in this instance is that it will _always_ be a string - what do we do in that case? Kind of renders a helper redundant no?

Reply:
well, assertions are for that, ensuring that something that should not happen does not happen
I think is reasonable when typings system comes a bit short